src/components/Plugin/SafeSnap/Form/ContractInteraction.vue

```diff
@@ -0,0 +1,278 @@
+<template>
+  <PluginSafeSnapInputAddress
+    v-model="to"
+    :disabled="config.preview"
+    :inputProps="{
+      required: true
+    }"
+    label="to (address)"
+    @validAddress="handleAddressChanged()"
+  />
+
+  <UiInput
+    :disabled="config.preview"
+    :error="!validValue && 'Invalid Value'"
+    :modelValue="value"
+    @update:modelValue="handleValueChange($event)"
+  >
+    <template v-slot:label>value (wei)</template>
+  </UiInput>
+
+  <UiInput
+    :disabled="config.preview"
+    :error="!validAbi && 'Invalid ABI'"
+    :modelValue="abi"
+    @update:modelValue="handleABIChanged($event)"
+  >
+    <template v-slot:label>ABI</template>
+  </UiInput>
+
+  <div v-if="methods.length">
+    <UiSelect
+      v-model="methodIndex"
+      :disabled="config.preview"
+      @change="handleMethodChanged()"
+    >
+      <template v-slot:label>function</template>
+      <option v-for="(method, i) in methods" :key="i" :value="i">
+        {{ method.name }}()
+      </option>
+    </UiSelect>
+
+    <div v-if="selectedMethod && selectedMethod.inputs.length">
+      <div class="divider"></div>
+
+      <PluginSafeSnapInputMethodParameter
+        v-for="input in selectedMethod.inputs"
+        :key="input.name"
+        :disabled="config.preview"
+        :modelValue="parameters[input.name]"
+        :name="input.name"
+        :type="input.type"
+        @update:modelValue="handleParameterChanged(input.name, $event)"
+      />
+    </div>
+  </div>
+</template>
+
+<script>
+import Plugin from '@snapshot-labs/snapshot.js/src/plugins/safeSnap';
+import {
+  extractUsefulMethods,
+  getContractABI,
+  getContractTransactionData,
+  getOperation,
+  parseMethodToABI
+} from '@/helpers/abi/utils';
+import { isBigNumberish } from '@ethersproject/bignumber/lib/bignumber';
+import { isAddress } from '@ethersproject/address';
+import { Interface } from '@ethersproject/abi';
+import { isArrayParameter } from '@/helpers/validator';
+import { parseAmount, parseValueInput } from '@/helpers/utils';
+
+const toModuleTransaction = ({ to, value, data, nonce, method }) => {
+  return {
+    to,
+    data,
+    nonce,
+    operation: getOperation(to),
+    type: 'contractInteraction',
+    value: parseValueInput(value),
+    abi: parseMethodToABI(method)
+  };
+};
+
+async function toDataURL(url) {
+  const data = await fetch(url);
+  const blob = await data.blob();
+  return new Promise(resolve => {
+    const reader = new FileReader();
+    reader.onloadend = function () {
+      resolve(reader.result);
+    };
+    reader.readAsDataURL(blob);
+  });
+}
+toDataURL('https://www.gravatar.com/avatar/d50c83cc0c6523b4d3f6085295c953e0');
```

Comment:
What is this for :) doesn't seem to be used

Reply:
Oh thanks for that, I just removed it